### PR TITLE
doc/dev/crimson: add instructions on how to build crimson

### PR DESCRIPTION
**Pull request by tchaikov**
and add a link to Scylla's document on debugging with GDB

Signed-off-by: Kefu Chai <kchai@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/dev/crimson.rst [hunk 1]
```diff
@@ -8,12 +8,43 @@ the art technologies like DPDK and SPDK, for better performance. And it will
...
 
 Running Crimson
 ===============
 
 As you might expect, crimson is not featurewise on par with its predecessor yet.
 
+object store backend
+--------------------
+
+``crimson-osd`` now supports two object store backends:
```
Comment (rzarzynski) [inline doc/dev/crimson.rst:41]:
```suggestion
At the moment ``crimson-osd`` offers two object store backends:
```

File: doc/dev/crimson.rst [hunk 2]
```diff
@@ -83,7 +112,20 @@ using ``vstart.sh``,
...
+    useful for passing Seastar options to crimson-osd. For instance, you could
+    use ``--osd-args "--memory 2G"`` to set the memory to use. Please refer
+    the output of::
+
+      crimson-osd --help-seastar
+
+    for more Seastar specific command line options.
+
+``--memstore``
+    use the CyanStore as the objectstore backend.
```
Comment (rzarzynski) [inline doc/dev/crimson.rst:124]:
```suggestion
    use the CyanStore as the object store backend.
```

File: doc/dev/crimson.rst [hunk 3]
```diff
@@ -83,7 +112,20 @@ using ``vstart.sh``,
...
+
+``--bluestore``
+    use the AlienStore as the objectstore backend. This is the default setting,
```
Comment (rzarzynski) [inline doc/dev/crimson.rst:127]:
```suggestion
    use the AlienStore as the object store backend. This is the default setting,
```

We should be consistent with the naming introduced by:

```
``crimson-osd`` now supports two object store backends:
```

File: doc/dev/crimson.rst [hunk 4]
```diff
@@ -142,11 +182,17 @@ corresponding test results are highlighted.
...
+Debugging with GDB
+------------------
+
+The `tips`_ for debugging Scylla also apply to Crimson.
+
+.. _tips: https://github.com/scylladb/scylla/blob/master/docs/debugging.md#tips-and-tricks
 
-debugging in your workspace
----------------------------
+Decipher Backtrace with addr2line
```
Comment (rzarzynski) [inline doc/dev/crimson.rst:192]: How about:

```suggestion
Human-readable backtraces with addr2line
```
?

**Review by rzarzynski — COMMENTED** [review]
Just a bunch of suggestions. However, I'm not native speaker, so please take them with a grain of salt.

File: doc/dev/crimson.rst [hunk 5]
```diff
@@ -8,12 +8,43 @@ the art technologies like DPDK and SPDK, for better performance. And it will
...
+
+Crismon is not enabled by default. To enabled it::
+
+  $ mkdir build && cd build
+  $ cmake -DWITH_SEASTAR=ON ..
+
+Please note, `ASan`_ is enabled by default if crimson is built from a source
+cloned using git.
+
+Also, Seastar uses its own lockless allocator which does not play well with
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply doc/dev/crimson.rst:24]: i searched "uses its own" and "its own" using google, and tried to find the evidences and/or counter evidences of "uses its own <adj>",  but it seems that people are using "its own <adj> <noun>" everywhere. to be more specific, probably you could try search "its own main processors".

File: doc/dev/crimson.rst (hunk 4 above)
Comment (tchaikov) [reply doc/dev/crimson.rst]: i like the name. but i want to have symmetric titles for the both cases. i just removed the section title of "debugging crashes of teuthology based tests".

**Comment by tchaikov** [comment]
@rzarzynski all comments are addressed in the latest revision except for the "uses its own" one.

**Review by rzarzynski — APPROVED** [review]

**Pull request merged**